app/controllers/staff/training_sessions_controller.rb

```diff
@@ -0,0 +1,91 @@
+class Staff::TrainingSessionsController < StaffAreaController
+  before_action :current_training_session, only: [:show, :edit, :update, :certify_trainees, :delete_training_session]
+  before_action :changed_params, only: [:update]
+
+  layout 'staff_area'
+
+  def index
+
+  end
+
+  def new
+    @new_training_session = TrainingSession.new
+  end
+
+  def create
+    @training_session = TrainingSession.new(training_session_params)
+    if params['training_session_users'].present?
+      JSON.parse(params['training_session_users']).each do |user|
+        @training_session.users << User.find_by(username: user)
+      end
+    end
+    if @training_session.save
+      redirect_to "#{staff_training_session_path(@training_session.id)}"
+    else
+      redirect_to :back
+      flash[:alert] = "Something went wrong. Please try again."
+    end
+  end
+
+  def update
+    #binding.pry
+    @current_training_session.update(changed_params)
+    if params['users'].present?
+      JSON.parse(params['users']).each do |new_user|
+        unless @current_training_session.users.include? User.find_by(username: new_user)
+          @current_training_session.users << User.find_by(username: new_user)
+        end
+      end
+      @current_training_session.users.each do |old_user|
+        unless JSON.parse(params['users']).include? old_user.username
+          @current_training_session.users.delete(old_user)
+        end
+      end
+    end
+
+    if @current_training_session.save
+      flash[:notice] = "Training session updated succesfully"
+    end
+    redirect_to :back
+  end
+
+  def certify_trainees
+    staff = current_user
+    if params['training_session_graduates'].present?
+      params['training_session_graduates'].each do |graduate|
+         if @current_training_session.users.include? User.find(graduate)
+           unless User.find(graduate).certifications.include? Certification.find_by(training: @current_training_session.training.name)
```

Comment:
This kind of uniqueness checking should live in the model. Something like this: https://stackoverflow.com/questions/8965089/validate-uniqueness-of-in-association